src/Simulation/CsharpGeneration.Tests/SimulationCodeTests.fs

```diff
@@ -3207,3 +3206,229 @@ namespace Microsoft.Quantum.Tests.LineNumbers
+[assembly: SpecializationDeclaration("{\"Kind\":{\"Case\":\"QsBody\"},\"TypeArguments\":{\"Case\":\"Null\"},\"Information\":{\"Characteristics\":{\"Case\":\"EmptySet\"},\"InferredInformation\":{\"IsSelfAdjoint\":false,\"IsIntrinsic\":false}},\"Parent\":{\"Namespace\":\"Microsoft.Quantum.Tests.UnitTests\",\"Name\":\"UnitTest2\"},\"Attributes\":[],\"SourceFile\":\"%%%\",\"Position\":{\"Item1\":25,\"Item2\":4},\"HeaderRange\":{\"Item1\":{\"Line\":1,\"Column\":11},\"Item2\":{\"Line\":1,\"Column\":20}},\"Documentation\":[]}")]
+
+#line hidden
+namespace Microsoft.Quantum.Core
+{
+    public class Attribute : UDTBase<QVoid>, IApplyData
+    {
+        public Attribute() : base(default(QVoid))
+        {
+        }
+
+        public Attribute(QVoid data) : base(data)
+        {
+        }
+
+        System.Collections.Generic.IEnumerable<Qubit> IApplyData.Qubits => null;
+        public void Deconstruct()
+        {
+        }
+    }
+}
+
+#line hidden
+namespace Microsoft.Quantum.Diagnostics
+{
+    public class Test : UDTBase<String>, IApplyData
+    {
+        public Test() : base(default(String))
+        {
+        }
+
+        public Test(String data) : base(data)
+        {
+        }
+
+        System.Collections.Generic.IEnumerable<Qubit> IApplyData.Qubits => null;
+        public void Deconstruct()
+        {
+        }
+    }
+}
+
+#line hidden
+namespace Microsoft.Quantum.Tests.UnitTests
+{
+    [SourceLocation("%%%", OperationFunctor.Body, 22, 26)]
+    public partial class UnitTest1 : Operation<QVoid, QVoid>, ICallable
+    {
+        public UnitTest1(IOperationFactory m) : base(m)
+        {
+        }
+
+        public class QuantumSimulator
```

Comment:
A different approach we could have used is to instead of creating a subclass per simulator, we could have created one single `Tests` class and use `[Xunit.Theory]` accepting the `Simulator` in the constructor and creating the instances using the `[Xunit.InlineData(new QuantumSimulator())]`. This might be a little cleaner and group the tests better.

Reply:
BTW, I'm ok keeping the generation as-is, just putting the idea as that might be cleaner.

Reply:
I like the idea, but it looks like IndlineData attribute (like any attribute) is not happy about the non-constant expression "new QuantumSimulator()"